Asked to waitFor but it was never even start()'ed - since 2.4.0

First, thanks for the nice project, it's of great help!

I'm using it for writing JUnit test. Lately i was upgrading from 2.3.0 to 2.4.0 and got those sporadic failures:
```
Caused by: ch.vorburger.exec.ManagedProcessException: An error occurred while running a command: create database if not exists `db1`;
at ch.vorburger.mariadb4j.DB.run(DB.java:300)
at ch.vorburger.mariadb4j.DB.run(DB.java:265)
at ch.vorburger.mariadb4j.DB.run(DB.java:269)
at ch.vorburger.mariadb4j.DB.createDB(DB.java:308)
at europa.testsupport.MySqlEnv.createDatabase(MySqlEnv.java:74)
... 78 more
Caused by: ch.vorburger.exec.ManagedProcessException: Asked to waitFor Program [/tmp/europa.domain.space.ListJdbcTablesHandlerTest5747555168995803376/mysql-base/bin/mysql, --socket=/tmp/MariaDB4j.35895.sock] (in working directory /tmp/europa.domain.space.ListJdbcTablesHandlerTest5747555168995803376/mysql-base), but it was never even start()'ed!
at ch.vorburger.exec.ManagedProcess.assertWaitForIsValid(ManagedProcess.java:478)
at ch.vorburger.exec.ManagedProcess.waitForExitMaxMsWithoutLog(ManagedProcess.java:439)
at ch.vorburger.exec.ManagedProcess.waitForExit(ManagedProcess.java:418)
at ch.vorburger.mariadb4j.DB.run(DB.java:298)
```
Those happended on the build server which is on linux. On my local Mac i hadn't had those problems.
Downgrading to 2.3.0 stabilized everything!

> I'm facing this issue too. But it's not reproducible, the next run usually succeeds.
> 2.4.0
> 
> > ch.vorburger.exec.ManagedProcessException: An error occurred while running a script file sourced from the classpath at: schema.sql
> > Caused by: ch.vorburger.exec.ManagedProcessException: Asked to waitFor Program [/tmp/MariaDB4j/base/bin/mysql, --socket=/tmp/MariaDB4j.3406.sock] (in working directory /tmp/MariaDB4j/base), but it was never even start()'ed!
> 
> This is strange, because I'm starting the maria database manually within a `@Rule`, thus this state should not happen at all!
> 
> ```
> public class MyRule implements BeforeAllCallback, AfterAllCallback {
> 	private MariaDB4jSpringService db;
> 
> 	@Override
> 	public void beforeAll(ExtensionContext extensionContext) throws Exception {
> 		db = new MariaDB4jSpringService();
> 		db.setDefaultPort(port);
> 		db.start();
> 		
> 		db.getDB().createDB("mytable");
> 		db.getDB().source("schema.sql");
> 	}
> 
> 	@Override
> 	public void afterAll(ExtensionContext extensionContext) {
> 		db.stop();
> 	}
> }
> ```
> 
> Usage:
> 
> ```
> @SpringBootTest
> public class TestClass {
> 	@RegisterExtension
> 	static MyRule db = new MyRule();
> }
> ```

This issue is caused by some MariaDB concurrent initial code. Two possible approaches, first try to use singleton initial in global OS (maybe not only the JVM), second try to use 2.3

I ran into this error again, only in a different part of the code (`ch.vorburger.mariadb4j.DB#run(java.lang.String, java.lang.String, java.lang.String, java.lang.String, boolean, boolean`):

```
ch.vorburger.exec.ManagedProcessException: An error occurred while running a script file sourced from the classpath at: mariadb_create_tables.sql
	at ch.vorburger.mariadb4j.DB.run(DB.java:371)
	at ch.vorburger.mariadb4j.DB.source(DB.java:306)
	at ch.vorburger.mariadb4j.DB.source(DB.java:287)
	at ch.vorburger.mariadb4j.DB.source(DB.java:256)
...
Caused by: ch.vorburger.exec.ManagedProcessException: Asked to waitFor Program [/tmp/MariaDB4j/base/bin/mysql, -Ddbname, --socket=/tmp/MariaDB4j.38401.sock] (in working directory /tmp/MariaDB4j/base), but it was never even start()'ed!
	at ch.vorburger.exec.ManagedProcess.assertWaitForIsValid(ManagedProcess.java:477)
	at ch.vorburger.exec.ManagedProcess.waitForExitMaxMsWithoutLog(ManagedProcess.java:437)
	at ch.vorburger.exec.ManagedProcess.waitForExit(ManagedProcess.java:416)
	at ch.vorburger.mariadb4j.DB.run(DB.java:369)
	... 20 more
```

I think this is occurring because of these lines from the `run()` method above:

```
            ManagedProcess process = builder.build();
            process.start();
            process.waitForExit();
```            
and may be an inherent race condition in ManagedProcess itself. The check that's failing in `ManagedProcess.waitForExit()`/`assertWaitForIsValid()` is:

```
if (!this.isAlive() && !this.resultHandler.hasResult())
``` 

Diving into `isAlive()`, it seems to only be set to true if, at the end of `startExecute()`, 

```
this.isAlive = this.watchDog.isWatching();
```

but `watchDog.isWatching()` is only set in `org.apache.commons.exec.ExecuteWatchdog.start()`, which in turn is only called by `DefaultExecutor.executeInternal()`. The version of `DefaultExecutor.execute(final CommandLine command, final Map<String, String> environment, final ExecuteResultHandler handler)` you're using spawns a **separate thread** to call `executeInternal`, and so the call stack from `process.start();` may not complete before the `process.waitForExit();` is checked. In particular in `DefaultExecutor.executeInternal()`, is the main call out to launch the process, which could sometimes take a while to complete (hence this error occurring intermittently for people?):

```
Process process = this.launch(command, environment, dir);
```
(see: https://commons.apache.org/proper/commons-exec/apidocs/src-html/org/apache/commons/exec/DefaultExecutor.html)

--- a/mariaDB4j-core/pom.xml
+++ b/mariaDB4j-core/pom.xml
@@ -24,7 +24,7 @@
         <dependency>
             <groupId>ch.vorburger.exec</groupId>
             <artifactId>exec</artifactId>
-            <version>3.1.3</version>
+            <version>3.1.4</version>
         </dependency>
         <dependency>
             <groupId>org.apache.commons</groupId>

